include/klee/ArrayExprOptimizer.h

```diff
@@ -0,0 +1,91 @@
+#ifndef KLEE_EXPROPTIMIZER_H
+#define KLEE_EXPROPTIMIZER_H
+
+#include <cassert>
+#include <iterator>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "Expr.h"
+#include "ExprBuilder.h"
+#include "Constraints.h"
+#include "Internal/Support/ErrorHandling.h"
+#include "util/ArrayExprVisitor.h"
+#include "util/Assignment.h"
+#include "util/Ref.h"
+#include "klee/AssignmentGenerator.h"
+#include "klee/ExecutionState.h"
+
+#include "llvm/Support/TimeValue.h"
+#include "klee/Internal/System/Time.h"
+
+#include <ciso646>
+
+#ifdef _LIBCPP_VERSION
+#include <unordered_map>
+#include <unordered_set>
+#define unordered_map std::unordered_map
+#define unordered_set std::unordered_set
+#else
+#include <tr1/unordered_map>
```

Comment:
We are using C++11, do we still need the check for tr1?

Reply:
Still required on my machine